core/wasmWasi/src/internal/TimeZonesInitializer.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019-2023 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.internal
+
+import kotlinx.datetime.IllegalTimeZoneException
+
+@RequiresOptIn
+internal annotation class InternalDateTimeApi
+
+/*
+This is internal API which is not intended to use on user-side.
+ */
+@InternalDateTimeApi
+public interface TimeZonesProvider {
+    public fun  zoneDataByName(name: String): ByteArray
+    public fun getTimeZones(): Set<String>
+}
+
+/*
+This is internal API which is not intended to use on user-side.
+ */
+@InternalDateTimeApi
+public fun initializeTimeZonesProvider(provider: TimeZonesProvider) {
+    check(timeZonesProvider != provider) { "TimeZone database redeclaration" }
+    timeZonesProvider = provider
+}
+
+@InternalDateTimeApi
+private var timeZonesProvider: TimeZonesProvider? = null
+
+private val utcData get() = byteArrayOf(
+    84,   90,  105,  102,   50,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,
+    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,
+    0,    0,    0,    0,    0,    0,    0,    1,    0,    0,    0,    1,    0,    0,    0,    0,
+    0,    0,    0,   84,   90,  105,  102,   50,    0,    0,    0,    0,    0,    0,    0,    0,
+    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,
+    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    1,    0,    0,    0,    4,    0,
+    0,    0,    0,    0,    0,   85,   84,   67,    0,   10,   85,   84,   67,   48,   10,
+)
+
+private val utcNames get() = setOf("Universal", "Etc/Universal", "Etc/Zulu", "Etc/UCT", "Etc/UTC", "Zulu", "UCT", "UTC")
```

Comment:
No-no, you don't need to do any of this: we already have special handling for `UTC` in the `TimeZone` initialization, it just needs to be present in the `availableTimeZones` set. Sorry for not being clear.

Though now that I look into it, you can even ignore that requirement, since on the JS, this rule also isn't enforced. We'll need to solve this in a centralized way.

Reply:
Ahh, so I roll it back